### PR TITLE
Make CC toolchains available for other targets

### PR DESCRIPTION
**Pull request by cpuguy83**
This cross-compiling other binaries simpler.
It would be nice if the cross Makefile target built all the required
bins, but at least this is a first step.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -94,26 +94,35 @@ RUN /download-frozen-image-v2.sh /build \
 	hello-world:latest@sha256:be0cd392e45be79ffeffa6b05338b98ebb16c87b255f48e297ec7f98e123905c
 # See also ensureFrozenImagesLinux() in "integration-cli/fixtures_linux_daemon_test.go" (which needs to be updated when adding images to this list)
 
-# Just a little hack so we don't have to install these deps twice, once for runc and once for dockerd
-FROM base AS runtime-dev-cross-false
+FROM base AS cross-false
+
+FROM cross-false AS cross-true
```
Comment (thaJeztah) [inline Dockerfile]: Might be clearer to just use `base` here

```suggestion
FROM base AS cross-true
```

File: Dockerfile [hunk 2]
```diff
@@ -94,26 +94,35 @@ RUN /download-frozen-image-v2.sh /build \
...
+RUN dpkg --add-architecture armhf
+RUN dpkg --add-architecture arm64
+RUN dpkg --add-architecture armel
+RUN if [ "$(go env GOHOSTARCH)" = "amd64" ]; then \
+	apt-get update \
+	&& apt-get install -y \
+		crossbuild-essential-armhf \
+		crossbuild-essential-arm64 \
+		crossbuild-essential-armel; \
```
Comment (thaJeztah) [inline Dockerfile]: Should we add a `--no-install-recommends` here?

**Review by thaJeztah — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (cpuguy83) [reply Dockerfile]: Sure.

**Comment by cpuguy83** [comment]
Updated this to add a hack to manually install amd64 runtime-dev packages in the cross image instead of inheriting since otherwise the v0 builder is broken.

File: Dockerfile (hunk 1 above)
Comment (cpuguy83) [reply Dockerfile]: Done.

**Review by thaJeztah — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
Janky failing on a flaky test;

```
23:39:19 FAIL: docker_api_swarm_test.go:292: DockerSwarmSuite.TestAPISwarmLeaderElection
```

I'll restart

**Review by yongtang — APPROVED** [review]
LGTM

**Pull request merged**